torch/_inductor/ir.py

```diff
@@ -2573,7 +2592,94 @@ def is_channels_last_stride_ordered(self):
         order = [len(order)] + order
         return self.is_stride_ordered(order)
 
+    @staticmethod
+    def _pad_strides(in_strides, size, align=DEFAULT_ALIGN):
+        """
+        The padding does not change stride order but makes sure all strides larger
+        than the threshold are multiple of align.
+        """
+        if len(in_strides) == 0:
+            return in_strides
+
+        # a loose check for channels last
```

Comment:
Should we check whether or not `layout_opt` is True to distinguish between whether or not we are considering `size[1] == 1` as channels last or not ?

Reply:
I just refactored this code to call `Layout.is_channels_last_contiguous` as Jason suggested above. We now check for channels last more accurately.

I think this check makes sense no matter if `layout_opt` is on or off.